subsys/mgmt/mcumgr/grp/fs_mgmt/src/fs_mgmt.c

```diff
@@ -338,6 +339,8 @@ static int fs_mgmt_file_download(struct smp_streamer *ctxt)
 
 end:
 	rc = (ok ? MGMT_ERR_EOK : MGMT_ERR_EMSGSIZE);
+
+unlock:
```

Comment:
Seems that the label is only useful when CONFIG_MCUMGR_GRP_FS_FILE_ACCESS_HOOK=y and compiler does not like that. You need to find a way around that.